### ISSUE TITLE
Shared flutter runtime

### ISSUE THREAD

**Issue by FelschR** [flutter/flutter#15784]
We all know that building a flutter app comes with some overhead in file size.
Is there a plan to ship the flutter core/runtime with the OS so that it can be shared between installed apps?

In the case of Android & Fuchsia it should be possible considering they're both also maintained by Google.
The most important part of that integration would be that the flutter framework can be updated Independent from the OS itself. Similar to how Google Play Services are working on Android.

**Comment by Hixie** [comment]
This isn't something we're planning on doing any time soon, though I agree it's an attractive idea.

The real problem is that it would tie us down to an unchanging engine, which is a very high cost. Having an unchanging blob of code on the device is a problem we've heard developers complain about when it comes to writing apps on Android across versions. Also, we currently update the engine multiple times a day, and someone could modify the engine and ship that. We'd like to make Flutter apps have much lower overhead even without having separately shipped the engine, ideally.

**Comment by FelschR** [comment]
As I've said a good approach to this issue would be making the flutter engine similar to the Google Play Services.
So that it auto-updates in the background or can be manually updated via the Play Store.

The only drawback I can think of is when 2 apps require 2 different versions of the flutter engine that aren't compatible.
But even that could still save a lot actual app size.
And I think via semver (or additional flagging) it could be easy to mark which flutter versions are compatible with another so that the system could always install the latest compatible versions.

**Comment by Hixie** [comment]
The problem with autoupdating the engine is that it would mean that apps change behaviour depending on what version is available, and if the wrong version is available, the app won't work, etc.

Obviously the size wins are there, we're just hoping to find another way to make the engine overhead smaller, such as by "just" making the engine actually smaller.

**Comment by crossle** [comment]
Good idea, other question? how to make flutter runtime to host native app, and just download a mini-program run Flutter app?

**Comment by pedromassangocode** [comment]
Labeling new feature based on the thread above.

**Comment by jonahwilliams** [comment]
We don't plan to do this.
